librad/src/net/peer.rs

```diff
@@ -76,17 +70,37 @@ pub struct BindError {
 pub struct Peer {
     key: SecretKey,
     paths: Paths,
+    git: GitStorage,
 }
 
 impl Peer {
-    pub fn new(paths: Paths, key: SecretKey) -> Self {
-        Self { key, paths }
+    pub fn new(paths: Paths, git: GitStorage) -> Self {
+        Self {
+            key: git.key.clone(),
+            paths,
+            git,
+        }
+    }
+
+    pub fn init(paths: Paths, key: SecretKey) -> Result<Self, git::storage::Error> {
+        let git = GitStorage::init(&paths, key.clone())?;
+        Ok(Self { key, paths, git })
+    }
+
+    pub fn peer_id(&self) -> PeerId {
+        PeerId::from(&self.key)
     }
 
     pub fn public_key(&self) -> PublicKey {
         self.key.public()
     }
 
+    // FIXME: this should not be here, but we can't otherwise do entity signing
+    // in tests
+    pub fn key(&self) -> &SecretKey {
```

Comment:
Can't we put it under `#[cfg(test)]`?

Reply:
Unfortunately no: integration tests (those in the `tests/`) don't run with with `cfg(test)` enabled apparently.

I though that perhaps we will eventually have a method `Peer::sign<T>(meta: &mut Entity<T>) -> ??`, but we need to equip the peer with a resolver for this.